frontend/routes/app/app.component.tsx

```diff
@@ -111,6 +113,21 @@ export class App extends React.PureComponent<IProps, IState> {
 				this.setState({ referrer: DEFAULT_REDIRECT });
 			});
 		}
+
+		if (isAuthenticated) {
+			// @ts-ignore
+			window.LC_API = window.LC_API || {};
```

Comment:
Could you move this logic to a new service called `livechat`? It'll be small service, but it'll be easier to manage livechat code. It's similar (or even the same) approach that you used adding analytics to the app.

For example:
```
// services/livechat.ts

const initialise = ( { firstName = '' , lastName = '' , email = '' } = {}) => {
	const name = `${firstName} ${lastName}`.trim();
	// @ts-ignore
	window.LC_API.set_visitor_name(name);
	// @ts-ignore
	window.LC_API.set_visitor_email(email);
};

export const LiveChat = { initialise };
```

Reply:
I didnt wanted to add angular code. But I'll move it to a helper.

Reply:
The other thing that I could do is create a react component with the livechat code, binded to the login state and setting the license from the config.

Reply:
@sanmont3drepo I didn't mean about angular service. I thought something like a helper (look at files in `services` directory) to handle external service. I also think that creating a new component is unnecessary. During the migration of top app component, we'll create a new module to set up things like this one during initialising process. So, probably, we'll have to remove this component and create service instead of it.

Reply:
@simplypixi  I tried today to convert it to just a service I need the props with the updated currentuser to tackle the async nature of configuring livechat. So its easier to have a component with the state binded as well as the license config. 
In the case of the service, its still needed to add the livechat from 'react-livechat' to some part of the react tree, so you wouldnt be able to get rid of the react tag so easy, unless we want to implement the import of the livechat script ourselves.

In the currentstate is just adding the custom livechat component to any place of the tree and that's it.

If you still have the opinion that it would be a simpler integration with the service approach, could you please expand more on the idea?

Reply:
I thought about something like this:
```
// services/livechat.ts
window.LC_API = window.LC_API || {};

const initialise = ( { firstName = '' , lastName = '' , email = '' } = {}) => {
	const name = `${firstName} ${lastName}`.trim();
	// @ts-ignore
	window.LC_API.set_visitor_name(name);
	// @ts-ignore
	window.LC_API.set_visitor_email(email);
};

export const LiveChat = { initialise };


// app.component.tsx
import LiveChat from 'react-livechat`;
import LiveChatService from '../.../services/livechat';

class App {
    /* ... */
    componentDidUpdate() {
        if (isAuthenticated) {
             LiveChatService.initialise(this.props.currentUser)
        }
    }

   render() {
        /* ... */
        <LiveChat licence={123} />
   }
}
```

It's ok if you think it's an overcomplicated solution. We can figure out something different when the proper time comes. Now, it can stay as is - I mean the component is also good solution (for now), but (probably) we'll have to remove it in the future.

Reply:
Yeah, the problem is that that solution doesnt work, because you have to tackle the cases when window.LC_API has not being initialized (the .on_after_load handler, which in that context you need the reference to props to have the updated current user,  because if you use just currentuser and bind the initilize function to that you get the object of the previous state of current user)
and in the case that window.LC_API has being initialized  you need o just run it for now also taking in consideration that the mail has being set, because it might take a little to fetch the whole user data.
That's why it cant be that simple. I wish it could.

Reply:
Got it ;)